<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>

<body>
    <script>

        class OddNumberError extends Error {
            constructor(message) {
                super(message);
                this.name = "OddNumberError";
            }
        }

        function somar(n1, n2, cb) {
            setTimeout(() => {
                const resultado = n1 + n2;

                if (resultado % 2 !== 0) {
                    cb(new OddNumberError("O valor é ímpar"));
                } else {
                    cb(null, resultado);
                }
            }, 2000);
        }

        somar(2, 3, (error, resultado) => {
            if (error) {
                console.log(`A operação falhou! ${error.message}`);
            } else {
                console.log(`A operação foi bem sucedida! O número ${resultado} é par`);
            }
        });





      




        class OddNumberError extends Error {
            constructor(message) {
                super(message);
                this.name = "OddNumberError";
            }
        }

        function somar(numero, cb) {
            setTimeout(() => {
                const resultado = n1 + n2;

                if (resultado % 2 !== 0) {
                    cb(new OddNumberError("O valor é ímpar"));
                } else {
                    cb(null, resultado);
                }
            }, 2000);
        }

       
        const resultado = dobrar(numero);
            console.log(`O dobro de ${numero} é ${resultado}`);
        } catch (error) {
            console.log(error.message);
        }

        console.log("Finalizando programa...");




















































    </script>
</body>

</html>